geoapps_utils/base.py

```diff
@@ -0,0 +1,384 @@
+                out_dict.update({key: value})
+
+        return out_dict
+
+    def flatten(self) -> dict:
+        """
+        Flatten the parameters to a dictionary.
+
+        :return: Dictionary of parameters.
+        """
+        out = self._recursive_flatten(self.model_dump())
+        out.pop("input_file", None)
+
+        return out
+
+    @property
+    def input_file(self) -> InputFile:
+        """Create an InputFile with data matching current parameter state."""
+
+        if self._input_file is None:
+            ifile = self._create_input_file_from_attributes()
+        else:
+            ifile = copy(self._input_file)
+            ifile.validate = False
+
+        return ifile
+
+    def _create_input_file_from_attributes(self) -> InputFile:
+        """
+        Create an InputFile with data matching current parameter state.
+        """
+        # ensure default uijson (PAth )exists or raise an error
+        if self.default_ui_json is None or not self.default_ui_json.exists():
+            raise FileNotFoundError(
+                f"Default uijson file '{self.default_ui_json}' not a valid path."
+            )
+
+        ifile = InputFile.read_ui_json(self.default_ui_json, validate=False)
+
+        attributes = self.flatten()
+
+        ifile.data = {
+            key: attributes.get(key, value) for key, value in ifile.data.items()
+        }
+
+        return ifile
+
+    def write_ui_json(self, path: Path) -> None:
+        """
+        Write the ui.json file for the application.
+
+        :param path: Path to write the ui.json file.
+        """
+        self.input_file.write_ui_json(path.name, str(path.parent))
+
+    def serialize(self):
+        """Return a demoted uijson dictionary representation the params data."""
+
+        dump = self.model_dump(exclude_unset=True)
+        dump["geoh5"] = str(dump["geoh5"].h5file.resolve())
+        ifile = self.input_file
+        ifile.update_ui_values(self._recursive_flatten(dump))
+        assert ifile.ui_json is not None
+        options = ifile.stringify(ifile.demote(ifile.ui_json))
+
+        return options
+
+
+def fetch_driver_class(json_dict: str | Path | dict) -> type[Driver]:
+    """
+    Fetch the driver class from the ui.json 'run_command'.
+
+    :param filepath: Path to a ui.json file with a 'run_command' key.
+    """
+    # TODO Remove after deprecation of geoapps_utils.driver
+    from geoapps_utils.driver.driver import (  # pylint: disable=import-outside-toplevel, cyclic-import
+        BaseDriver,
+    )
+
+    if isinstance(json_dict, (str, Path)):
+        with open(json_dict, encoding="utf-8") as jsonfile:
+            uijson = load(jsonfile)
+    else:
+        uijson = json_dict
+
+    if not isinstance(uijson, dict) or "run_command" not in uijson:
+        raise ValueError(
+            f"Invalid ui.json file: {json_dict}. It must contain a 'run_command' key."
+        )
+
+    module = __import__(uijson["run_command"], fromlist=["Driver"])
```

Comment:
I think you can avoid the loop below by using the `importlib.import_module` wrapper as it returns the requested object instead of the top-level package (https://docs.python.org/3/library/importlib.html#importlib.import_module)